engines/router/missionctl/server/upi/server.go

```diff
@@ -40,9 +37,67 @@ func NewUPIServer(mc missionctl.MissionControlUPI) *Server {
 	}
 }
 
+// UniversalPredictionServiceServer is the internal server that communicates in bytes
+// It requires the codec to be passing []byte instead of a proto object. It is registered
+// the same way as the generated proto which has the following interface instead.
+// PredictValues(context.Context, *PredictValuesRequest) (*PredictValuesResponse, error)
+type UniversalPredictionServiceServer interface {
+	PredictValues(context.Context, []byte) ([]byte, error)
+}
+
+//nolint:revive,gofmt,goimports
+// UniversalPredictionServiceServiceDesc implement grpc internal methodHandler interface, it creates an io.writer
+// and expect the codec to write bytes into the input instead of the typical proto request.
+// Nolint is used as context cannot be changed to first argument in order to fit the interface and there are
+// other bugs on fmt and imports while using nolint
+func UniversalPredictionServiceServiceDesc(
+	srv interface{},
+	ctx context.Context,
+	dec func(interface{}) error,
+	interceptor grpc.UnaryServerInterceptor,
+) (interface{}, error) {
+	in := new(bytes.Buffer)
+	if err := dec(in); err != nil {
```

Comment:
this `dec` decode function is coded and pass in from grpc server. The data is unmarshal in the server.go via this dec func passed. Which is implemented [here](https://github.com/grpc/grpc-go/blob/master/server.go#L1294).

The unmarshal is done via codec here `if err := s.getCodec(stream.ContentSubtype()).Unmarshal(d, v);` from `d` which is the data, into `v`, which is `in` above. 

So there are only 2 ways to workaround this. 
1. For client to pass in contentSubtype(), i tried with various client (postman and golang header) with `content-type` = `application/grpc+fiber` as per the http2 grpc [definition](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md#requests), but it just doesn't get recognized on the server end. The only success I tried was a golang client using invoke with `callOption` `grpc.CallContentSubtype(codecName)`, but that would be too restrictive.

2. Is to enforce a default codec, this way any client call will invoke this client codec. Which set [this](https://github.com/grpc/grpc-go/blob/master/server.go#L1831) and will be returned in precedence of the proto default codec. 

The implementation for this is using 2.

Reply:
Technically what we can do is to check whether the content type is `application/grpc` and then not use the decoder altogether?